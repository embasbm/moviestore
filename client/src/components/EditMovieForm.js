import React, { Component } from 'react';
import { ControlLabel, FormGroup, FormControl, Button } from 'react-bootstrap';
class EditMovieForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      id: this.props.movie.id,
      title: this.props.movie.title,
      text: this.props.movie.text,
      category_id: this.props.movie.category_id
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleSubmit(e) {
    e.preventDefault();
    const { id, title, text, category_id } = this.state;
    this.props.editMovie(id, title, text, category_id);
  }

  render() {
    return (
      <td>
        <form onSubmit={this.handleSubmit}>
          <FormGroup>
            <ControlLabel>Title</ControlLabel>
            <FormControl
              name="title"
              inputRef={this.titleInput}
              id="movieTitle"
              type="text"
              label="Movie title"
              value={this.state.title}
              onChange={this.handleChange}
            />
            <ControlLabel>Text</ControlLabel>
            <FormControl
              name="text"
              inputRef={this.textInput}
              id="movieText"
              type="text"
              label="Movie text"
              componentClass="textarea"
              value={this.state.text}
              onChange={this.handleChange}
            />
            <ControlLabel>Category</ControlLabel>
            <FormControl
              name="category_id"
              componentClass="select"
              placeholder="select"
              inputRef={ref => { this.categoryId = ref; }}
              onChange={this.handleChange}
            >
              {this.props.categories && this.props.categories.map(category => {
                return (
                  <option key={category.id} value={category.id} selected={category.id === this.state.category_id ? true : false}>{category.name}</option>
                )
              })}
            </FormControl>
          </FormGroup>
          <Button type="submit">Update Movie</Button>
        </form>
      </td>
    )
  }
}
export default EditMovieForm;
