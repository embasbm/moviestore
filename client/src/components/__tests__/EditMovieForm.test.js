import React from 'react';
import axios from "axios";
const ReactTestRenderer = require('react-test-renderer');
import EditMovieForm from '../EditMovieForm';

jest.mock('axios');

describe('EditMovieForm', () => {
  const movie = {
    "id": 1,
    "title": "spiderman",
    "text": "Omnis repellat harum. Error beatae placeat. Ut minima nam."
  };

  const categories = [
    {
      id: 1,
      name: 'foo'
    }
  ]
  axios.get.mockImplementation(() => Promise.resolve(resp));

  it('Should compare EditMovieForm component with a snapshot', () => {
    const component = ReactTestRenderer.create(<EditMovieForm movie={movie} categories={categories} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
