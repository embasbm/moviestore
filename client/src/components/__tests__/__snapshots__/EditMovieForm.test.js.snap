// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMovieForm Should compare EditMovieForm component with a snapshot 1`] = `
<td>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Title
      </label>
      <input
        className="form-control"
        id="movieTitle"
        label="Movie title"
        name="title"
        onChange={[Function]}
        placeholder="spiderman"
        type="text"
      />
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Text
      </label>
      <textarea
        className="form-control"
        id="movieText"
        label="Movie text"
        name="text"
        onChange={[Function]}
        placeholder="Omnis repellat harum. Error beatae placeat. Ut minima nam."
        type="text"
      />
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Category
      </label>
      <select
        className="form-control"
        id={undefined}
        name="category_id"
        onChange={[Function]}
        placeholder="select"
        type={undefined}
      >
        <option
          selected={false}
          value={1}
        >
          foo
        </option>
      </select>
    </div>
    <button
      className="btn btn-default"
      disabled={false}
      type="submit"
    >
      Update Movie
    </button>
  </form>
</td>
`;
